R/rvar-.R

```diff
@@ -663,9 +791,62 @@ cleanup_draw_dims <- function(x) {
     rownames(x) <- as.character(seq_rows(x))
   }
 
+  # if x is factor-like (character or with "levels" attr), make it a factor
+  if (is.character(x)) {
```

Comment:
is it worth making this "as.factor" code its own function for easy reuse and extention?